### PR TITLE
Add syntax highlighting to README

### PR DESCRIPTION
**Pull request by adamchainz**
Makes it a bit easier to follow.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -9,39 +9,49 @@ and ``with_pattern()`` when ``import \*`` is used:
 
 From there it's a simple thing to parse a string:
 
->>> parse("It's {}, I love it!", "It's spam, I love it!")
-<Result ('spam',) {}>
->>> _[0]
-'spam'
+.. code-block:: pycon
```
Comment (jenisys) [inline README.rst]: Language should be "python" or "py", but not "pycon" (AFAIK).

IN ADDITION: 
`README.rst` is generated from `parse.py`.
tox runs doctests based on this information.
See "tox.ini".

Comment (adamchainz) [reply README.rst]: `pycon` is "python console": https://pygments.org/docs/lexers/#pygments.lexers.python.PythonConsoleLexer

**Comment by adamchainz** [comment]
Copied changes back into `parse.py`.

**Comment by r1chardj0n3s** [comment]
Thanks!!

**Pull request merged**